src/app/auth/login.tsx

```diff
@@ -19,45 +25,136 @@ function LoginScreen() {
     router.replace(path);
   };
 
-  const signInWithEmail = async () => {
+  const signIn = async () => {
     setLoading(true);
-    const { error } = await sessionHandler.signInWithEmail(email, password);
+    let email;
+    if (validator.isEmail(emailOrUsername)) {
+      email = emailOrUsername;
+
+      const { count } = await supabase
+        .from('profiles')
+        .select(`*`, { count: 'exact' })
+        .eq('email', email);
+
+      if (count == 0) {
+        setError(
+          'An account with that email does not exist. Please try again.',
+        );
```

Comment:
I think you need to add a setLoading(false) before the return here, because right now if I trigger this email error, when I edit the username/email/password fields the log in button remains disabled.